aep/general/0124/aep.md.j2

```diff
@@ -1,5 +1,86 @@
 # Resource association
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/57) for more
-information.
+APIs sometimes have resource hierarchies that can not be cleanly expressed in
+the usual tree structure. For example, a resource may have a many-to-one
+relationship with two other resource types instead of just one. Alternatively,
+a resource may have a many-to-many relationship with another resource type.
+
+## Guidance
+
+A resource **must** have at most one canonical parent, and `List` requests
+**must not** require two distinct "parents" to work.
+
+### Multiple many-to-one associations
+
+If a resource has a many-to-one relationship with multiple resource types, it
+**must** choose at most one of them to be the canonical parent. The resource
+**may** be associated with other resources through other fields on the
+resource.
+
+{% tab proto %}
+
+{% sample 'multiple_many_to_one.proto', 'message Book' %}
+
+{% tab oas %}
+
+{% sample 'multiple_many_to_one.oas.yaml', 'Book:' %}
+
+{% endtabs %}
+
+When listing resources with multiple associations in this way, the RPC **must**
+treat the `string parent` field as required as discussed in [AEP-132][], and
+**must not** add additional required arguments. The RPC **should** include a
+`string filter` field that allows users to filter by other resource
+associations as discussed in [AEP-160][].
+
+**Note:** Resource reference fields **must** accept the same resource name
+format that is used in the `name` field of the referenced resource.
+
+### Many-to-many associations
+
+Many-to-many associations are less common in APIs than they are in relational
+databases, in part because they are more difficult to model and present over
+network interfaces.
+
+An API **may** contain many-to-many relationships, and **should** use a
+repeated field containing a list of resource names, following the principles
+described for repeated fields in [AEP-144][].
+
+
+{% tab proto %}
+
+{% sample 'many_to_many_repeated.proto', 'message Book' %}
+
+{% tab oas %}
+
+{% sample 'many_to_many_repeated.oas.yaml', 'Book:' %}
+
+{% endtabs %}
+
+**Note:** See [AEP-144][] for more information on repeated fields, including
+how to handle common issues such as atomic changes.
+
+If the use of a repeated field is too restrictive, or if more metadata is
+required along with the association, an API **may** model a many-to-many
+relationship using a sub-resource with two one-to-many associations.
+
+{% tab proto %}
+
+{% sample 'many_to_many_subresource.proto', 'message BookAuthor' %}
+
+{% tab oas %}
+
+{% sample 'many_to_many_subresource.oas.yaml', 'BookAuthor:' %}
+
+{% endtabs %}
+
+**Note:** Using subresources to model an association between resources is only
+recommended if additional metadata is required in the relationship, or if the
+restrictions around the use of a repeated field preclude the use of that
```

Comment:
we don't have to block on this - but it's really not clear to me what this note is saying. Or maybe it's just lack of context?

What additional metadata is required such that we should model a resource relationship as a sub-resource?

What types of restrictions would there be around the use of a repeated field?

Reply:
I agree that the repeated field part is difficult to parse.

The additional metadata is a bit more intriguing. What if you had a `Publisher` enum on `BookAuthor` to denote if this was self-published, not yet published, or published by a major label? You would need that sub-resource to help denote that metadata

(I'm not fully convinced that that's the best way to mark that contrived relationship, but that's an aside)

Reply:
let's pick it up in https://github.com/aep-dev/aep.dev/issues/157, no need to block.